### PR TITLE
Task/add notif

### PR DESCRIPTION
**Pull request by xiaojiang990209**
- Add toast messages for api calls and favourite courses

### CODE REVIEW

File: client/src/components/Button/index.jsx [hunk 1]
```diff
@@ -1,9 +1,13 @@
...
   border-color: ${({theme}) => theme.color.primary}
 `;
 
-export default Button;
\ No newline at end of file
+export const CourseTitleButton = styled(Button)`
+  height: ${({theme}) => theme.unit.large}px;
+`
+
+export default Button;
```
Comment (yawenwinnie) [inline client/src/components/Button/index.jsx]: if Button is already default exported, it doe snot need to be exported again when initializing

File: client/src/ducks/matchable.js [hunk 2]
```diff
@@ -7,5 +8,9 @@ export const matchGroup = (maxMembers, courseID, date, hasTime) => {
 
 export const createGroup = (startDate, courseID, groupSize, duration, userID) => {
   const body = { groupSize, startDate, courseID, duration, userID };
-  return axios.post('/api/matchable/groups', body);
+  return withNotification(
+    'Group created successfully',
```
Comment (yawenwinnie) [inline client/src/ducks/matchable.js]: make the sentences constants inside ducks/matchable/constants, there should not be any hardcoded strings inside function calls

File: client/src/ducks/matchable.js [hunk 3]
```diff
@@ -7,5 +8,9 @@ export const matchGroup = (maxMembers, courseID, date, hasTime) => {
...
+    'Failed to create group',
+    'Creating group'
+  )(axios.post('/api/matchable/groups', body));
```
Comment (yawenwinnie) [inline client/src/ducks/matchable.js]: make the API a const inside the constants folder as above

File: client/src/utils/sendNotification.js [hunk 4]
```diff
@@ -0,0 +1,24 @@
+import toast from 'cogo-toast';
+
+const option = {
+  hideAfter: 1.5
+};
+
+export const showSuccessNotif = (message) => toast.success(message, option);
+
+export const showLoadingNotif = (message) => toast.loading(message, option);
```
Comment (yawenwinnie) [inline client/src/utils/sendNotification.js]: i don't see these functions used anywhere else except this folder, do we want to export it?

Comment (xiaojiang990209) [reply client/src/utils/sendNotification.js]: We need the `showSuccessNotif` when user favourites a course :)

**Pull request merged**